cmd/admin-handlers.go

```diff
@@ -862,13 +962,30 @@ func (a adminAPIHandlers) SetConfigKeysHandler(w http.ResponseWriter, r *http.Re
 			writeErrorResponseJSON(w, ErrAdminConfigBadJSON, r.URL)
 			return
 		}
-		// Calculate the type of the current key from the
-		// original config json
-		jsonFieldType := gjson.Get(configStr, k).Type
-		// Convert passed value to json filed type
-		val, cErr := convertValueType(elem, jsonFieldType)
-		if cErr != nil {
-			writeCustomErrorResponseJSON(w, ErrAdminConfigBadJSON, cErr.Error(), r.URL)
+
+		// Compare key against struct config
+		jsonField := gjson.Get(structConfig, k)
+		if (gjson.Result{}) == jsonField {
+			reqInfo := (&logger.ReqInfo{}).AppendTags("Config key", k)
+			// GetReq Info to keep the prev setting
+			ctx = logger.SetReqInfo(ctx, reqInfo)
+			logger.LogIf(ctx, errors.New("Invalid key"))
+			writeCustomErrorResponseJSON(w, ErrNoSuchKey, "Invalid KEY: "+k, r.URL)
+			return
+		}
+		if err, k2 := isInvalidJSONField(jsonField, elem); err != "" {
```

Comment:
can k2 be renamed so that this code is self-documenting

Reply:
Also, the style should be such that error is the second parameter being returned and not the first one.

Reply:
OK